nimble/host/src/ble_gap.c

```diff
@@ -1291,6 +1291,10 @@ ble_gap_rx_adv_set_terminated(struct hci_le_adv_set_terminated *evt)
 
     ble_gap_adv_finished(evt->adv_handle, reason, conn_handle,
                          evt->completed_events);
+
+    /* Remove ble_gap_slave instance */
+    memset(&ble_gap_slave[evt->adv_handle], 0, 
+           sizeof(struct ble_gap_slave_state));
```

Comment:
this is not valid,  set terminate means that advertising was stopped, not that instance was removed (ie you can re-start advertising without configuring instance)

Reply:
removed this.